### DISCUSSION TITLE
Topics for next meeting

### DISCUSSION THREAD

**Discussion by andreaa93** [FAIRmat-NFDI/AreaA-data_modeling_and_schemas#57] [Data Modeling Meeting]
Hello everyone! 

I've run through several things I would like to discuss with you guys and I list them here:

- there is a results.material.element and then elemental_composition.element how do we harmonize these duplicate?
- I have the first practical example of a sample collection class (MOVPE Ga2O3 IKZ use case), let's revise the class together and let's push it into our data modeling
- I need again to exchange about an activity modifying an entity, in the specific case the substrate proprocess. Should we create a new substrate archive or update the existing one? The user usually doesn't want a new ID after preprocessing the substrate, so we should update the existing archive
- About dopants in Substrates or in evaporated layers: should they be part of the elemental_composition?
- many times users are writing in their processes a sampleID, but not a real reference. How can we optimize this?

spamming tags: @hampusnasstrom @Pepe-Marquez @budschi

**Comment by hampusnasstrom** [comment]
Thanks for the good discussion points @aalbino2! Let's discuss these in the data modeling meeting on 2.8.2023. Here are my quick thoughts on them now:

> * there is a results.material.element and then elemental_composition.element how do we harmonize these duplicate?

They are intentionally duplicated as one just lists elements while the other is only filled if the fractions are known. These will both move into the topology section.

> * I have the first practical example of a sample collection class (MOVPE Ga2O3 IKZ use case), let's revise the class together and let's push it into our data modeling

Okay, for now I'm adding the following into [MR 1385](https://gitlab.mpcdf.mpg.de/nomad-lab/nomad-FAIR/-/merge_requests/1385):
```python
class SectionReference(ArchiveSection):
    '''
    A section used for referencing another section.
    '''
    name = Quantity(
        type=str,
        description='A short descriptive name for the role of this reference.',
        a_eln=ELNAnnotation(
            component='StringEditQuantity',
        )
    )
    reference = Quantity(
        type=ArchiveSection,
        description='A reference to a NOMAD archive section.',
        a_eln=ELNAnnotation(
            component='ReferenceEditQuantity',
            label='Section Reference',
        ),
    )


class ActivityReference(SectionReference):
    '''
    A section used for referencing an Activity.
    '''
    reference = Quantity(
        type=Activity,
        description='A reference to a NOMAD `Activity` entry.',
        a_eln=ELNAnnotation(
            component='ReferenceEditQuantity',
            label='Activity Reference',
        ),
    )


class EntityReference(SectionReference):
    '''
    A section used for referencing an Entity.
    '''
    reference = Quantity(
        type=Entity,
        description='A reference to a NOMAD `Entity` entry.',
        a_eln=ELNAnnotation(
            component='ReferenceEditQuantity',
            label='Entity Reference',
        ),
    )


class Experiment(Activity):
    '''
    A section for grouping activities together into an experiment.
    '''
    activities = SubSection(
        section_def=ActivityReference,
        description='References to the activities that make up the experiment.',
        repeats=True,
    )


class Collection(Entity):
    '''
    A section for grouping entities together into a collection.
    '''
    entities = SubSection(
        section_def=EntityReference,
        description='References to the entities that make up the collection.',
        repeats=True,
    )
```

> * I need again to exchange about an activity modifying an entity, in the specific case the substrate proprocess. Should we create a new substrate archive or update the existing one? The user usually doesn't want a new ID after preprocessing the substrate, so we should update the existing archive

Let's discuss it in the data modeling meeting next week but instinctively I would say the substrate should update itself in this case.

> * About dopants in Substrates or in evaporated layers: should they be part of the elemental_composition?

That was my intention, yes.

> * many times users are writing in their processes a sampleID, but not a real reference. How can we optimize this?

We could add a field `lab_id` to the `EntityReference` class above together with a normalizer that attempts to fill the references by querying for an entry with that `lab_id`. How does that sound?

**Comment by andreaa93** [comment]
Other points:
- Class and ReferenceClass detailed use
- syntax 1..1, 1..*
- Extend the Material Processing with Natascha and Ta-Shun classes
